File: src/main/kotlin/gg/match/domain/board/overwatch/entity/Overwatch.java
package gg.match.domain.board.overwatch.entity;

import gg.match.controller.common.entity.Expire;
import java.util.*;
import javax.persistence.*;

@Entity
@Table(name = "overwatch")
public class Overwatch {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @Enumerated(EnumType.STRING)
    private Tier tier;
    @Enumerated(EnumType.STRING)
    private Position position;
    private Boolean voice;
    private String content;
    @Enumerated(EnumType.STRING)
    private Expire expire;
    private Date regdate;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setTier(Tier tier) {
        this.tier = tier;
    }

    public Tier getTier() {
        return tier;
    }

    public void setPosition(Position position) {
        this.position = position;
    }

    public Position getPosition() {
        return position;
    }

    public void setVoice(Boolean voice) {
        this.voice = voice;
    }

    public Boolean getVoice() {
        return voice;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getContent() {
        return content;
    }

    public void setExpire(Expire expire) {
        this.expire = expire;
    }

    public Expire getExpire() {
        return expire;
    }

    public void setRegdate(Date regdate) {
        this.regdate = regdate;
    }

    public Date getRegdate() {
        return regdate;
    }
}

File: src/main/kotlin/gg/match/domain/board/overwatch/entity/Overwatch.kt
package gg.match.domain.board.overwatch.entity

import gg.match.common.entity.BoardBaseEntity
import gg.match.controller.common.entity.Expire
import gg.match.domain.board.overwatch.dto.HeroResponseDTO
import gg.match.domain.board.overwatch.dto.OverwatchRequestDTO
import gg.match.domain.board.overwatch.dto.ReadOverwatchBoardDTO
import javax.persistence.*

@Entity
@Table(name = "overwatch")
class Overwatch(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    var oauth2Id: String,

    @Enumerated(EnumType.STRING)
    var type: Type,

    @Enumerated(EnumType.STRING)
    var tier: Tier,

    @Enumerated(EnumType.STRING)
    var position: Position,

    @Column(length=1)
    var voice: String,

    var content: String,

    @Enumerated(EnumType.STRING)
    var expire: Expire,

    var expired: String = "false"

): BoardBaseEntity(){
    fun toReadOverwatchBoardDTO(hero: HeroResponseDTO, memberList: List<String>, banList: List<String>): ReadOverwatchBoardDTO {
        return ReadOverwatchBoardDTO(
            id = id,
            oauth2Id = oauth2Id,
            type = type,
            tier = tier,
            position = position,
            voice = voice,
            content = content,
            expire = expire,
            expired = expired,
            created = created,
            author = hero,
            chatRoomId = chatRoomId,
            memberList = memberList,
            banList = banList
        )
    }

    fun update(overwatchRequestDTO: OverwatchRequestDTO){
        type = overwatchRequestDTO.type
        tier = overwatchRequestDTO.tier
        position = overwatchRequestDTO.position
        voice = overwatchRequestDTO.voice
        content = overwatchRequestDTO.content
        expire = overwatchRequestDTO.expire
    }

    fun update(expired: String){
        this.expired = expired
    }
}